genai-perf/tests/test_checkpoint.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+from unittest.mock import patch
+
+from genai_perf.checkpoint.checkpoint import Checkpoint
+from genai_perf.config.input.config_command import ConfigCommand
+from genai_perf.config.run.results import Results
+from tests.test_utils import create_run_config
+
+
+class TestCheckpoint(unittest.TestCase):
```

Comment:
Yes, but for this I need to mock up the file I/O. Going to file a future story to do this.

Reply:
You can mock the calls using monkey patch. Some of our unit tests today do that. I don't want to start having unit tests added to the code which actually use I/O. That creates problems.

I know this is a feature branch, but if we're deferring this, I would want us to have some kind of check to make sure that it gets fixed before it goes into the main branch. It'd be easier to get it done as part of the PR, even if it's a track on a separate ticket.

Reply:
Due to time constraints and the requirement that this is checked in to the feature branch before I leave on PTO this is not a change I will be making at this time.

I have filed a story to mock the I/O and add more error checking and this will get completed before the feature branch is merged into main.

Reply:
Sounds good, thanks Brian. Appreciate it!